plugins/catalog-backend/src/ingestion/CatalogRules.ts

```diff
@@ -32,6 +32,9 @@ export type CatalogRule = {
     target?: string;
     type: string;
   }>;
+  sources?: Array<{
```

Comment:
fly-by review even though it's still draft: I'm think this introduces new names for things that already have established terminology, i.e. locations. It could be something like `locations.match` for example? I don't think it's worth the magic to omit the protocol, so it might be something like

```yaml
catalog:
  rules:
    - allow: [Component]
      locations:
      - match: https://github.com/backstage/*
```

There are many other ways to structure that though, and I think it's worth exploring that space a bit further.

Reply:
Thank you @Rugvip we liked the suggestion and made some changes. 

The new solution works but I think it conflicts a bit with an older commit and I wanted to get your opinion on that. 

right now there is a test in the project called `should not care about location configuration in catalog.rules`. It seems to expect the location value to be be ignored in global rules which conflicts with this. I updated the test to how it could work with just not having location key. Is this going to cause other issues?